staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/customresource_handler.go

```diff
@@ -747,7 +747,7 @@ func (d schemaCoercingDecoder) Decode(data []byte, defaults *schema.GroupVersion
 		}
 	}
 
-	return obj, gvk, nil
+	return obj.DeepCopyObject(), gvk, nil
```

Comment:
calling `DeepCopyObject` is for escaping type equality check [here](https://github.com/kubernetes/kubernetes/blob/323e1375b3b15a4144d46d41155ca402a2fcef63/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L106). this change is soooo hacky that we need to clearly document it and mark a `// TODO: Remove it after versioning decoder is fixed` here. but yeah, this pull is indeed a minimal walkaround. we could apply it before we have a better solution.

Reply:
only necessary if into != nil.

Reply:
It's even hackier than I thought: `Decode` here uses `into`, but returns the deep copy.  The converter then is called with the deep copy (as source) and the same `into` (as destination) [here](https://github.com/kubernetes/kubernetes/blob/323e1375b3b15a4144d46d41155ca402a2fcef63/staging/src/k8s.io/apimachinery/pkg/runtime/serializer/versioning/versioning.go#L126).  Hence, `into` still has the GVK set which makes the [GVK check in our converter](https://github.com/kubernetes/kubernetes/pull/68452/files#diff-b64fd00b8a8fc70715bdc95154eb114dR51) not fail.

Because source and destination have the same GVK, the [conversion call](https://github.com/kubernetes/kubernetes/pull/68452/files#diff-b64fd00b8a8fc70715bdc95154eb114dR60) is a no-op.

Finally, the `context.(runtime.GroupVersioner)` is used to do another conversion attempt.

OMG.